app_route/main.tf

```diff
@@ -0,0 +1,25 @@
+data "cloudfoundry_org" "org" {
+  name = var.cf_org_name
+}
+data "cloudfoundry_space" "space" {
+  name = var.cf_space_name
+  org  = data.cloudfoundry_org.org.id
+}
+
+data "cloudfoundry_domain" "domain" {
+  name = var.domain
+}
+
+locals {
+  destinations = (length(var.app_ids) == 0 ? null : [
+    for dest in var.app_ids : { app_id = dest }
+  ])
```

Comment:
this is actually probably the solution for using the application module within egress proxy as well.

Reply:
one of the blockers for using the application module by the egress proxy was that the proxy doesn't bind any services, and the `cloudfoundry_app` module rejects `service_bindings = []` as invalid. Using this ternary to actually set `service_bindings = null` will likely get around that